### ISSUE TITLE
Plugin doesn't ever start generating embeddings

### ISSUE THREAD

**Issue by Relic180** [brianpetro/obsidian-smart-connections#498]
Installed the plugin and start seeing notifications along the side prompting to "start generating embeddings".

I click the button to start and notification that says "Embedding smart_notes..." sits at 0. Checking CPU/Mem usage shows that not much is actually happening.

![embedding smart_notes...](https://i.ibb.co/RCXMKYg/Screenshot-2024-02-29-161451.png)

Checked console and there are a few errors during init of the plugin, and also after clicking to start the embedding.

![console errors](https://i.ibb.co/LS4RYT3/Screenshot-2024-02-29-161129.png)

Did try stopping and restarting plugin a couple times, as well as stopping and restarting Obsidian.

I also included my Open AI API key and received a notification that the API key was invalid, although it isn't. Console shows an error that the request to test the API key itself apparently failed. I don't know if this is related to the above issue or not, but including it just in case.

![api key error](https://i.ibb.co/CbZRj5D/Screenshot-2024-02-29-195628.png)

**Comment by brianpetro** [comment]
@Relic180 the 429 error your receiving is from rate limits. You need to add money to your OpenAI account to fix it.

🌴

**Comment by Relic180** [comment]
Are you sure about the rate limit? I checked my OpenAI panel and I it looks like I have 100% of my budget for the month, but I'm still seeing that error (It's reset for March but was the same for Feb).

![available api limit](https://i.ibb.co/S6Md3mR/Screenshot-2024-03-01-081734.png)

Anyhow, I removed the API completely and it does look like the console output changed slightly, but still throws an error.

![console log without api key](https://i.ibb.co/BrP74MH/Screenshot-2024-03-01-081744.png)

**Comment by liuling73** [comment]
I have the same problem here, the program keeps getting stuck there, how do I change the GPT‘s speed?

**Comment by brianpetro** [comment]
These errors where OpenAI is blocking/restricting your request are often solved by pre-paying $5 on your account.

If you still have the problem after that you might need to reach out to OpenAI customer support.

**Comment by Relic180** [comment]
Wow yea, just now realizing how wonkey OpenAI's billing model is. The the balance being separate from the limit, and have "auto-recharge" as a third setting.

Anyway, got that straightened out and apparently the embeddings are able to generate now. I didn't realize until now that I had selected an API-based embedding option. So apparently that was what I did wrong. I didn't even look for that, and since I didn't read closely enough I guess I assumed all the embeddings would be local.

Maybe there's a little wiggle room to improve the error notifications around that, maybe to help point that out?
